src/main/scala/chisel3/aop/Select.scala

```diff
@@ -391,7 +394,9 @@ object Select {
     val isPort = module._component.get
       .asInstanceOf[DefModule]
       .ports
-      .flatMap { p => DataMirror.collectAllMembers(p.id) }
+      .map(_.id)
+      .collect { case (p: Data) => p }
+      .flatMap { p => DataMirror.collectAllMembers(p) }
```

Comment:
```suggestion
      .flatMap {
        case Port(d: Data, _, _) => DataMirror.collectAllMembers(d)
        case _ => Nil
      }
```
I don't think this code is particularly performant to begin with, but this did just turn 1 traverse and copy of a sequence into 3